src/full/core/log.cljc

```diff
@@ -0,0 +1,157 @@
+
+     (defmacro debug
+       "Logs args with debug loglevel. Returns nil."
+       [& args]
+       `(if-cljs
+          (.debug js/console (full.core.log/format-log [~@args]))
+          (log/debug context ~@args)))
+
+     (defmacro info
+       "Logs args with info loglevel. Returns nil."
+       [& args]
+       `(if-cljs
+          (.info js/console (full.core.log/format-log [~@args]))
+          (log/info context "XXX" ~@args "YYY")))
+
+     (defmacro warn [& args]
+       `(if-cljs
+          (.warn js/console (full.core.log/format-log [~@args]))
+          (log/warn context ~@args)))
+
+     (defmacro error [x & more]
+       `(if-cljs
+          (.error js/console (full.core.log/format-log [~x ~@more]))
+          (let [x# ~x]
+            (if (instance? Throwable x#)
+              (log/error x# context ~@more)
+              (log/error context x# ~@more)))))
+
+     (defmacro level-enabled?
+       "Checks if log level is enabled."
+       [level]
+       `(if-cljs
+          true
+          (log/enabled? ~level)))
+
+     (defmacro group
+       "Begins grouped log."
+       [& args]
+       `(if-cljs
+          (if (.-groupCollapsed js/console)
+            (.groupCollapsed js/console (full.core.log/format-log [~@args]))
+            ; fallback for older browsers
+            (.log js/console (full.core.log/format-log [~@args])))
+          (do)))
+
+     (defmacro group-end
+       "Ends grouped log. ClojureScript only."
+       []
+       `(if-cljs
+          (when (.-groupEnd js/console)
+            (.groupEnd js/console))
+          (do)))
+
+       ;;; Configuration. Clojure only.
+
+     (defn check-config-file [config-file]
+       (let [f (as-file config-file)]
+         (when (not (.exists f))
+           (println "full.core.log - EXITING!- Log configuration file"
+                    (.getAbsolutePath f)
+                    "not found.")
+           (System/exit 1))
+         (println "full.core.log - Using log config file" (.getAbsolutePath f))
+         f))
+
+     (defn configure
+       ; TODO: If we make this as a macro, would it be possible to load this
+       ;       module without the logback dependency?
+       ([] (configure @log-config))
+       ([config-file]
+        (let [context (org.slf4j.LoggerFactory/getILoggerFactory)]
+          (try
+            (let [configurator (ch.qos.logback.classic.joran.JoranConfigurator.)]
+              (.setContext configurator context)
+              (.reset context)
+              (.doConfigure configurator (as-url (check-config-file config-file))))
+            (catch Exception _))  ; StatusPrinter will handle this
+          (ch.qos.logback.core.util.StatusPrinter/printInCaseOfErrorsOrWarnings
+            context))))
+     ))
+
+(defn do-info
+  "Evaluates all arguments and logs them with info loglevel. Returns the value
+   of the last argument."
+  [& args]
+    (info (string/join ", " args))
+    (last args))
+
+(defn do-debug
```

Comment:
Will both `do-info` and `do-debug` work in cljs? The `info` and `debug` are defined only for Clojure.